### PR TITLE
Update Readme with config and usage information

### PR DESCRIPTION
**Pull request by brendandeere**
The readme now contains how to configure the gem and make/process a simple subscription

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -19,6 +19,60 @@ bundle
...
+-----
+This gem requires a gateway which supports credit cards in order to process
+subscription orders. By default the gem will look for the first active gateway
+with Spree::CreditCard as its `payment_source_class`.
+
+If you would like to specify the gateway used by the gem you can add this to
+an initialzer.
+
+```ruby
+  SolidusSubscriptions::Config.default_gateway = my_gateway
```
Comment (alexblackie) [inline README.md]: indentation here looks weird

File: README.md [hunk 2]
```diff
@@ -19,6 +19,60 @@ bundle
 bundle exec rails g solidus_subscriptions:install
 ```
 
+Configuration
+-----
```
Comment (alexblackie) [inline README.md]: I'm pretty sure we prefer the inline hierarchy for headings, i.e., `## Configuration`

File: README.md [hunk 3]
```diff
@@ -19,6 +19,60 @@ bundle
...
+```ruby
+  SolidusSubscriptions::Config.default_gateway = my_gateway
+```
+
+Usage
+-----
+
+### Purchasing Subscriptions
+By default only Spree::Variants are subscribable. To subscribe to a variant, it
+must have the `:subscribable` attrubute set to true.
```
Comment (alexblackie) [inline README.md]: >attribute

File: README.md [hunk 4]
```diff
@@ -19,6 +19,60 @@ bundle
...
+      quantity: 2,           // number of units in each subscription order,
+      subscribable_id: 1234, // Which variant the subscription is for,
+      interval: 2592000      // Time between subscription orders (in seconds... because Ruby),
+      max_installments: 12   // Stop processing after this many subscription orders
+                             // (use null to process the subscription ad nauseam)
+    }
+  }
+```
+
+This will add associate a `SolidusSubscriptions::LineItem` to the line item
```
Comment (alexblackie) [inline README.md]: maybe
>This will associate a `SolidusSubscriptions::LineItem` to the `Spree::LineItem` being added...

File: README.md [hunk 5]
```diff
@@ -19,6 +19,60 @@ bundle
...
+being added to the cart.
+
+The customer will not be charged for the subscription until it is processed. The
+subscription line items should be shown to the user on the cart page by
+looping over `Spree::Order#subscription_line_items`.
+
+When the order is finalized, a full `SolidusSubscriptions::Subscription` will be
+created for each subscription line items associated to the order (through line items).
```
Comment (alexblackie) [inline README.md]: > for each subscription line item associated to the order, through the order's line items.

File: README.md [hunk 6]
```diff
@@ -19,6 +19,60 @@ bundle
...
+
+### Processing Subscriptions
+
+To process actionable subscriptions simply run:
+
+`bundle exec rake solidus_subscriptions:process`
+
+To schedule this task we suggest using the [whenever](https://github.com/javan/whenever) gem
```
Comment (alexblackie) [inline README.md]: probably want to capitalise Whenever and add a full-stop to the end of this sentence.

File: README.md (hunk 2 above)
Comment (brendandeere) [inline README.md]: Yeah, I was following what was already in the generated readme. But I can update it.

File: README.md (hunk 3 above)
Comment (brendandeere) [inline README.md]: lol #auto-correct-for-vim

File: README.md (hunk 2 above)
Comment (brendandeere) [inline README.md]: Yeah I was following what was in the generated file. But I will update all of them

File: README.md (hunk 4 above)
Comment (brendandeere) [inline README.md]: :+1:

File: README.md (hunk 6 above)
Comment (brendandeere) [inline README.md]: :+1:

File: README.md (hunk 5 above)
Comment (brendandeere) [inline README.md]: :+1:

**Comment by brendandeere** [comment]
Review fixed, but gh reviews are terrible until they implement the outdated review feature

**Comment by brendandeere** [comment]
@alexblackie this is ready for review again :)

**Pull request merged**